### ISSUE TITLE
IISAdministration module fails to load under SSH remoting

### ISSUE THREAD

**Issue by jhkimnew** [PowerShell/PowerShell#9498]
# Steps to reproduce

```
1.	Install 19h1 OS where we made IISAdministration module to be compatible to Powershell Core
2.	Install IIS
3.	Enable sshd feature with running below powershell commands:
Add-WindowsCapability -Online -Name OpenSSH.Server~~~~0.0.1.0
Set-Service -Name sshd -StartupType 'Automatic' 
Start-Service sshd
4.	Update the existing Subsystem setting of \ProgramData\ssh\sshd_config file as the following:
#Subsystem              sftp          sftp-server.exe
Subsystem powershell               pwsh.exe -NoLogo -NoProfile -sshs
5.	Restart sshd service 
6.	Install Powershell Core
7.	Start Powershell Core and then import iisadministration module via ssh remoting with running below commands:
Pwsh.exe
Enter-PSSession -HostName localhost -UserName administrator
Import-module iisadministration -verbose
```

# Expected behavior

```
No error and Microsoft.IIS.Powershell.Commands.dll in GAC should not be loaded to pwsh.exe process and the Microsoft.IIS.Powershell.Commands.dll in module directory should be used.
```

# Actual behavior

```
Import-module cmdlet shows  “Assembly with same name is already loaded” error message because Microsoft.IIS.Powershell.Commands.dll in GAC is loaded to pwsh.exe process. 

NOTE: 
1.	If you can’t reproduce in initial time, restart pwsh.exe and try again and you will see the issue. 
FYI, sometimes, the issue is not reproducible in the first trial. I guess that it is because the Microsoft.IIS.Powershell.Commands.dll is not in GAC for the first trial.
2.	If we delete the Microsoft.IIS.Powershell.Commands.dll in GAC, the issue is gone. 
That means this issue happens because pwsh.exe loads Microsoft.IIS.Powershell.Commands.dll in GAC directory unexpectedly when it starts. 
You can confirm that that with attaching the pwsh process right after it starts or simply running “tlist -p <process id of pwsh>” command and you will see the GAC directory path as the following:
C:\> tlist 6904 
   6904 pwsh.exe
   CWD:     C:\WINDOWS\system32\
   CmdLine: pwsh
…
10.0.18890.1001 sh  0x000001F99B180000  C:\WINDOWS\Microsoft.NET\assembly\GAC_MSIL\Microsoft.IIS.Powershell.Commands\v4.0_10.0.0.0__31bf3856ad364e35\Microsoft.IIS.PowerShell.Commands.dll
…

```

# Environment data

```none

```

**Comment by SteveL-MSFT** [comment]
@adityapatwardhan can you look at this one?

**Comment by chuanjiao10** [comment]
hi @adityapatwardhan , i confirm this , Unable to reproduce the problem.

pscode:
```
$服务器ip = '192.168.1.201'
$服务器上的账户名 = 'xxx'
[scriptblock]$测试命令 =
{
	get-childitem d:\
} 

$连接2 = New-PSSession -HostName $服务器ip -UserName $服务器上的账户名
#这里将提示你输入密码
$要返回的内容 = Invoke-Command -ScriptBlock $测试命令 -Session $连接2
$要返回的内容
$要返回的内容.name  # dir 返回的文件名

Invoke-Command -ScriptBlock {Import-module iisadministration -verbose;pause } -Session $连接2
```
return:
PS /root> 
PS /root> Invoke-Command -ScriptBlock {Import-module iisadministration -verbose;pause } -Session $连接2
VERBOSE: Loading module from path 'C:\Windows\system32\WindowsPowerShell\v1.0\Modules\iisadministration\iisadministration.psd1'.
VERBOSE: Loading 'TypesToProcess' from path 'C:\Windows\system32\WindowsPowerShell\v1.0\Modules\iisadministration\iiscommands.types.ps1xml'.
VERBOSE: Loading 'FormatsToProcess' from path 'C:\Windows\system32\WindowsPowerShell\v1.0\Modules\iisadministration\iiscommands.format.ps1xml'.
VERBOSE: Loading module from path 'C:\Windows\system32\WindowsPowerShell\v1.0\Modules\iisadministration\Microsoft.IIS.Powershell.Commands.dll'.
VERBOSE: Importing cmdlet 'Get-IISAppPool'.
VERBOSE: Importing cmdlet 'Start-IISCommitDelay'.
VERBOSE: Importing cmdlet 'Stop-IISCommitDelay'.
VERBOSE: Importing cmdlet 'Get-IISSite'.
VERBOSE: Importing cmdlet 'Remove-IISSite'.
VERBOSE: Importing cmdlet 'New-IISSite'.
VERBOSE: Importing cmdlet 'Stop-IISSite'.
VERBOSE: Importing cmdlet 'Start-IISSite'.
VERBOSE: Importing cmdlet 'Get-IISConfigSection'.
VERBOSE: Importing cmdlet 'Get-IISConfigCollection'.
VERBOSE: Importing cmdlet 'Get-IISConfigCollectionElement'.
VERBOSE: Importing cmdlet 'New-IISConfigCollectionElement'.
VERBOSE: Importing cmdlet 'Remove-IISConfigCollectionElement'.
VERBOSE: Importing cmdlet 'Remove-IISConfigElement'.
VERBOSE: Importing cmdlet 'Clear-IISConfigCollection'.
VERBOSE: Importing cmdlet 'Get-IISConfigElement'.
VERBOSE: Importing cmdlet 'Get-IISConfigAttributeValue'.
VERBOSE: Importing cmdlet 'Remove-IISConfigAttribute'.
VERBOSE: Importing cmdlet 'Set-IISConfigAttributeValue'.
VERBOSE: Importing cmdlet 'Get-IISServerManager'.
VERBOSE: Importing cmdlet 'Reset-IISServerManager'.
VERBOSE: Importing cmdlet 'Get-IISCentralCertProvider'.
VERBOSE: Importing cmdlet 'Set-IISCentralCertProvider'.
VERBOSE: Importing cmdlet 'Set-IISCentralCertProviderCredential'.
VERBOSE: Importing cmdlet 'Clear-IISCentralCertProvider'.
VERBOSE: Importing cmdlet 'Disable-IISCentralCertProvider'.
VERBOSE: Importing cmdlet 'Enable-IISCentralCertProvider'.
VERBOSE: Importing cmdlet 'Enable-IISSharedConfig'.
VERBOSE: Importing cmdlet 'Disable-IISSharedConfig'.
VERBOSE: Importing cmdlet 'Export-IISConfiguration'.
VERBOSE: Importing cmdlet 'Get-IISSharedConfig'.
VERBOSE: Importing cmdlet 'Get-IISSiteBinding'.
VERBOSE: Importing cmdlet 'Remove-IISSiteBinding'.
VERBOSE: Importing cmdlet 'New-IISSiteBinding'.
VERBOSE: Loading module from path 'C:\Windows\system32\WindowsPowerShell\v1.0\Modules\iisadministration\iisadministration.psm1'.
VERBOSE: Exporting cmdlet 'Get-IISAppPool'.
VERBOSE: Exporting cmdlet 'Start-IISCommitDelay'.
VERBOSE: Exporting cmdlet 'Stop-IISCommitDelay'.
VERBOSE: Exporting cmdlet 'Get-IISSite'.
VERBOSE: Exporting cmdlet 'Remove-IISSite'.
VERBOSE: Exporting cmdlet 'New-IISSite'.
VERBOSE: Exporting cmdlet 'Stop-IISSite'.
VERBOSE: Exporting cmdlet 'Start-IISSite'.
VERBOSE: Exporting cmdlet 'Get-IISConfigSection'.
VERBOSE: Exporting cmdlet 'Get-IISConfigCollection'.
VERBOSE: Exporting cmdlet 'Get-IISConfigCollectionElement'.
VERBOSE: Exporting cmdlet 'New-IISConfigCollectionElement'.
VERBOSE: Exporting cmdlet 'Remove-IISConfigCollectionElement'.
VERBOSE: Exporting cmdlet 'Remove-IISConfigElement'.
VERBOSE: Exporting cmdlet 'Clear-IISConfigCollection'.
VERBOSE: Exporting cmdlet 'Get-IISConfigElement'.
VERBOSE: Exporting cmdlet 'Get-IISConfigAttributeValue'.
VERBOSE: Exporting cmdlet 'Remove-IISConfigAttribute'.
VERBOSE: Exporting cmdlet 'Set-IISConfigAttributeValue'.
VERBOSE: Exporting cmdlet 'Get-IISServerManager'.
VERBOSE: Exporting cmdlet 'Reset-IISServerManager'.
VERBOSE: Exporting cmdlet 'Get-IISCentralCertProvider'.
VERBOSE: Exporting cmdlet 'Set-IISCentralCertProvider'.
VERBOSE: Exporting cmdlet 'Set-IISCentralCertProviderCredential'.
VERBOSE: Exporting cmdlet 'Clear-IISCentralCertProvider'.
VERBOSE: Exporting cmdlet 'Disable-IISCentralCertProvider'.
VERBOSE: Exporting cmdlet 'Enable-IISCentralCertProvider'.
VERBOSE: Exporting cmdlet 'Enable-IISSharedConfig'.
VERBOSE: Exporting cmdlet 'Disable-IISSharedConfig'.
VERBOSE: Exporting cmdlet 'Export-IISConfiguration'.
VERBOSE: Exporting cmdlet 'Get-IISSharedConfig'.
VERBOSE: Exporting cmdlet 'Get-IISSiteBinding'.
VERBOSE: Exporting cmdlet 'Remove-IISSiteBinding'.
VERBOSE: Exporting cmdlet 'New-IISSiteBinding'.
VERBOSE: Importing cmdlet 'Clear-IISCentralCertProvider'.
VERBOSE: Importing cmdlet 'Clear-IISConfigCollection'.
VERBOSE: Importing cmdlet 'Disable-IISCentralCertProvider'.
VERBOSE: Importing cmdlet 'Disable-IISSharedConfig'.
VERBOSE: Importing cmdlet 'Enable-IISCentralCertProvider'.
VERBOSE: Importing cmdlet 'Enable-IISSharedConfig'.
VERBOSE: Importing cmdlet 'Export-IISConfiguration'.
VERBOSE: Importing cmdlet 'Get-IISAppPool'.
VERBOSE: Importing cmdlet 'Get-IISCentralCertProvider'.
VERBOSE: Importing cmdlet 'Get-IISConfigAttributeValue'.
VERBOSE: Importing cmdlet 'Get-IISConfigCollection'.
VERBOSE: Importing cmdlet 'Get-IISConfigCollectionElement'.
VERBOSE: Importing cmdlet 'Get-IISConfigElement'.
VERBOSE: Importing cmdlet 'Get-IISConfigSection'.
VERBOSE: Importing cmdlet 'Get-IISServerManager'.
VERBOSE: Importing cmdlet 'Get-IISSharedConfig'.
VERBOSE: Importing cmdlet 'Get-IISSite'.
VERBOSE: Importing cmdlet 'Get-IISSiteBinding'.
VERBOSE: Importing cmdlet 'New-IISConfigCollectionElement'.
VERBOSE: Importing cmdlet 'New-IISSite'.
VERBOSE: Importing cmdlet 'New-IISSiteBinding'.
VERBOSE: Importing cmdlet 'Remove-IISConfigAttribute'.
VERBOSE: Importing cmdlet 'Remove-IISConfigCollectionElement'.
VERBOSE: Importing cmdlet 'Remove-IISConfigElement'.
VERBOSE: Importing cmdlet 'Remove-IISSite'.
VERBOSE: Importing cmdlet 'Remove-IISSiteBinding'.
VERBOSE: Importing cmdlet 'Reset-IISServerManager'.
VERBOSE: Importing cmdlet 'Set-IISCentralCertProvider'.
VERBOSE: Importing cmdlet 'Set-IISCentralCertProviderCredential'.
VERBOSE: Importing cmdlet 'Set-IISConfigAttributeValue'.
VERBOSE: Importing cmdlet 'Start-IISCommitDelay'.
VERBOSE: Importing cmdlet 'Start-IISSite'.
VERBOSE: Importing cmdlet 'Stop-IISCommitDelay'.
VERBOSE: Importing cmdlet 'Stop-IISSite'.
Press Enter to continue...: 
PS /root> 

Environment data
-------------------------
client : centos 7.6 + ps 6.2 
win 10 vm 18362.53

**Comment by adityapatwardhan** [comment]
@jhkimnew Could you confirm which PowerShell Core version you used?

**Comment by jhkimnew** [comment]
@adityapatwardhan 

I used Powershell 6.2.0. I also sent a mail to you with the repro machine information so that you can reproduce the issue.

**Comment by jhkimnew** [comment]
@chuanjiao10 Did you try more than one time? The issue is not reproducible in the first trial and so you would need to retry one more time with restarting pwsh.exe.

BTW, I used the built-in SSHD in OS, not public build of OpenSSH.

**Comment by chuanjiao10** [comment]
need to retry one more time with restarting pwsh.exe ???
see my ps code,i use command in a ps session.
at session "$连接2", not need install-module again ,no need start pwsh.exe ,not need restart pwsh.exe.

**Comment by jhkimnew** [comment]
@chuanjiao10 For some reasons, Microsoft.IIS.Powershell.Commands.dll in GAC is not loaded to pwsh.exe process and the problem is not reproducible. However, the issue happens when restarting pwsh.exe. That sounds strange but that is the symptom of this issue. BTW, I gave @adityapatwardhan the repro machine. So, he will be able to see the symptom of this issue. :-)

**Comment by adityapatwardhan** [comment]
I investigated this issue and it is due to [StartupProfileOptimization](https://docs.microsoft.com/en-us/dotnet/api/system.runtime.profileoptimization?view=netframework-4.8&viewFallbackFrom=netcore-2.1) done by dotnet. We need an ability to disable the profile optimization to avoid loading the assembly from GAC.

Details:

When we use the IISAdministration module in the SSH server, dotnet adds it to the profile optimization file. So when the next time the SSH server is started, it tries to load the DLL. Now since the PowerShell Module is not being loaded, we fallback to looking up the GAC for loading.

**Comment by iSazonov** [comment]
@SteveL-MSFT @adityapatwardhan @anmenaga It seems it is something expected to being fixed for 7.0.
